validator/client/src/test/java/tech/pegasys/teku/validator/client/VoluntaryExitDataProviderTest.java

```diff
@@ -39,4 +57,49 @@ void calculateCurrentEpoch_shouldReturnEpoch() {
     final UInt64 expectedEpoch = UInt64.valueOf(2083);
     assertThat(epoch).isEqualTo(expectedEpoch);
   }
+
+  @Test
+  void createSignedVoluntaryExit_shouldReturnSignedVoluntaryExit() {
+    final BLSSignature signature = dataStructureUtil.randomSignature();
+    final Validator activeValidator = getValidator(signature);
+    final int validatorIndex = 0;
+    when(keyManager.getActiveValidatorKeys()).thenReturn(List.of(activeValidator));
+
+    final UInt64 epoch = dataStructureUtil.randomEpoch();
+    final ForkInfo forkInfo = dataStructureUtil.randomForkInfo();
+
+    final SignedVoluntaryExit signedVoluntaryExit =
+        provider.createSignedVoluntaryExit(
+            validatorIndex, activeValidator.getPublicKey(), epoch, forkInfo);
+
+    final SignedVoluntaryExit expected =
+        new SignedVoluntaryExit(
+            new VoluntaryExit(epoch, UInt64.valueOf(validatorIndex)), signature);
+    assertThat(signedVoluntaryExit).isEqualTo(expected);
+  }
+
+  @Test
+  void createSignedVoluntaryExit_throwsExceptionWhenValidatorNotInList() {
+    final BLSSignature signature = dataStructureUtil.randomSignature();
+    final Validator activeValidator = getValidator(signature);
```

Comment:
You probably don't need an active validator to test this path. getActiveValidatorKeys() can return an empty list.